version/info_go118.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.18
+// +build go1.18
+
+package version
+
+import "runtime/debug"
+
+var computedRevision string
+
+func getRevision() string {
+	return computedRevision
```

Comment:
Can we do something like this so that users that customize the Revision at build time (long sha, short sha, non-git revisions) can keep their formatting?

```suggestion
	if Revision != "" {
		return Revision
	}
	return computedRevision
```

Reply:
For example, I remember that we explicitly didn't use the git hash for Revision for GitLab for CI artifact caching reasons.